sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -58,7 +58,7 @@ def decode_eventgrid_event(self, eventgrid_event, **kwargs): # pylint: disable=n
             eventgrid_event = EventGridEvent._from_json(eventgrid_event, encode) # pylint: disable=protected-access
             deserialized_event = EventGridEvent.deserialize(eventgrid_event)
             EventGridEvent._deserialize_data(deserialized_event, deserialized_event.event_type) # pylint: disable=protected-access
-            return deserialized_event
+            return cast(EventGridEvent, deserialized_event)
```

Comment:
same question

Reply:
this is necessary unlike above - we get
```
Returning Any from function declared to return "EventGridEvent"
```

because the msrest's deserialize method on line 59 isn't typed

Reply:
Then let's fix autorest too :)
https://github.com/Azure/msrest-for-python/issues/226

But ok here, since we should not wait for msrest fix